### ISSUE TITLE
Stuff needed for 1.13

### ISSUE THREAD

**Issue by Kobata** [Inventory-Tweaks/inventory-tweaks#578]
- [ ] All the coremod-dependent stuff needs redone, particularly the container parts. For now, I'm not going to be worried about redoing all of it as coremods: The only part that really *needs* to be is the hack for keyboard focus and it may be possible to work around that a different way, and if not it's a relatively simple patch anyway.
- [ ] LWJGL3-related stuff, particularly the input stuff (See #575)
- [ ] Pull out the tool-class stuff and just replace it with Forge's `ToolType` (See comments in #575)
- [ ] More? Probably want to investigate if there's been any added new APIs that could simplify things.

**Comment by josephrooks** [comment]
Nothing helpful to contribute but just wanted to say that I came back to Minecraft recently after a few years, and InvTweaks was the only mod I remembered or missed. Appreciate your work!

**Comment by JFGTX** [comment]
The same, with some friends we gone on 1.13, and every time I open my inventory, or a chest, I think about IT 😍 That mod I really simple, but awesome and unforgettable when you used it one time!
Waiting 1.13 😍😍😍

**Comment by Qwerty-Space** [comment]
Any update?

**Comment by darkklegend** [comment]
hi,

I know you wait for a better version of Forge for the 1.13.2, but have you try your mod with " Rift " ?

**Comment by jadar** [comment]
Just submitted #592. I am willing to help with the update process, if you'd be willing to delegate a couple of tasks. Don't want to redo work that has already been done, and you probably have a roadmap planned out in your head.

**Comment by FormSans** [comment]
Any update? I love this mod and need it desperately

**Comment by im-griff** [comment]
Could we have a bleeding edge build for 1.13 at least?

> Man, I really wish I had inventory tweaks for 1.13.2 right now
-Mahatma Gandhi

**Comment by CyanSlinky** [comment]
Please update for 1.14 the alternatives are broken and sub-par, this is the best mod and it's hard to live without it.

**Comment by Qwerty-Space** [comment]
I found two good, working alternatives.  But it doesn't have as many features:

https://www.curseforge.com/minecraft/mc-mods/item-scroller

https://www.curseforge.com/minecraft/mc-mods/inventory-sorting
https://github.com/kyrptonaught/Inventory-Sorter

**Comment by CyanSlinky** [comment]
Yeah that's what i meant with sub-par and all of those are fabric mods that i can't use with forge, and the only forge mod has broken sorting. I really want inventory tweaks on 1.14 since it actually works and has proper sorting rules etc.

**Comment by i509VCB** [comment]
Now that 1.15 is a thing I am considering possibly taking up the project (most likely a complete rewrite) and do some abstraction so that I can make it run on Forge and Fabric (a new mod system since 1.14 snapshots.

Now a reason for me adopting another platform is we can in theory get Invtweaks on snapshots with Fabric. 

I'll start trying to rewrite it all.

**Comment by mkalam-alami** [comment]
@i509VCB Supporting this Fabric API for fast releases seems like a good idea. Reminds me of the good old ModLoader era, back when I could update and release InvTweaks within a couple days of a new Minecraft version 😄 Keep us updated!
